com.unity.netcode.gameobjects/Components/Interpolator/BufferedLinearInterpolator.cs

```diff
@@ -0,0 +1,202 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Unity.Netcode
+{
+    /// <summary>
+    /// Solves for incoming values that are jittered
+    /// Partially solves for message loss. Unclamped lerping helps hide this, but not completely
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    internal abstract class BufferedLinearInterpolator<T> where T : struct
+    {
+        // interface for mock testing, abstracting away external systems
+        public interface IInterpolatorTime
```

Comment:
Does this need to be public?

Reply:
Hehe... I was reading up on this and saw this one person who was evidently looking at this very scenario:

> An internal interface may declare a public property of an internal type. A public type may implement this internal interface if it is done explicitly.
> 
> Technically, this allows the public type to have a public property of an internal type. However, the interface is internal and so the internal type is not actually exposed (it is not possible for external code to reference the technically-public property).

Sort of made my brain jiggle a little when I read that...

Reply:
mmm thinking of it, even if we set the parent class public, I wouldn't want this interface exposed. Switching to internal